examples/read_nodes.rs

```diff
@@ -0,0 +1,115 @@
+// example that separates logic for reading different top-level nodes of xml tree
+
+use quick_xml::events::{BytesStart, Event};
+use quick_xml::reader::Reader;
+use std::collections::HashMap;
+use std::str;
+
+// Code doesn't work if DefaultSettings node is constructed like this:
+//   <DefaultSettings Language="es" Greeting="HELLO"/>
+
+const XML: &str = r#"
+<?xml version="1.0" encoding="utf-8"?>
+  <DefaultSettings Language="es" Greeting="HELLO">
+  </DefaultSettings>  
+  <Localization>
+    <Translation Tag="HELLO" Language="ja">
+        <Text>こんにちは</Text>
+    </Translation>
+    <Translation Tag="BYE" Language="ja">
+        <Text>さようなら</Text>
+    </Translation>
+     <Translation Tag="HELLO" Language="es">
+         <Text>Hola</Text>
+     </Translation>
+     <Translation Tag="BYE" Language="es">
+         <Text>Adiós</Text>
+     </Translation>
+  </Localization>
+"#;
+
+#[derive(Debug)]
+struct Translation {
+    tag: String,
+    lang: String,
+    text: String,
+}
+
+impl Translation {
+    fn new_from_element(
+        reader: &mut Reader<&[u8]>,
+        element: BytesStart,
+    ) -> Result<Translation, quick_xml::Error> {
+        use std::borrow::Cow;
+        let mut tag = Cow::Borrowed("");
+        let mut lang = Cow::Borrowed("");
+        // let (tag, lang) =
+        for attr_result in element.attributes() {
+            let a = attr_result.unwrap();
+            match a.key.as_ref() {
+                b"Language" => lang = a.unescape_value()?,
+                b"Tag" => tag = a.unescape_value()?,
+                _ => (),
+            }
+        }
+        let mut text = Cow::Borrowed("");
+        let mut element_buf = Vec::new();
+        let event = reader.read_event_into(&mut element_buf)?;
+        match event {
+            Event::Text(ref e) => {
+                text = str::from_utf8(e.as_ref())?.into();
```

Comment:
That code has two problems:
- it will not work if you decide to enable `encoding` feature
- it does not decode entities, such as `&lt;`

It is better to use `e.unescape()?`:
```suggestion
                text = e.unescape()?;
```